test/lit/passes/remove-unused-brs.wast

```diff
@@ -340,9 +340,139 @@
+  ;; CHECK-NEXT:     (call $get-i32)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $restructure-br_if-value-redundant-in-block-tail-2 (result i32)
+    ;; As above, but now the value is different, so we do not optimize
+    (block $parent (result i32)
+      (call $nothing)
+      (drop
+        (br_if $parent
+          (i32.const 2)
+          (call $get-i32)
+        )
+      )
+      (i32.const 1)
+    )
+  )
+
+  ;; CHECK:      (func $restructure-br_if-value-redundant-in-block-tail-3 (type $0) (param $x i32) (result i32)
+  ;; CHECK-NEXT:  (block $parent (result i32)
+  ;; CHECK-NEXT:   (call $nothing)
+  ;; CHECK-NEXT:   (drop
+  ;; CHECK-NEXT:    (br_if $parent
+  ;; CHECK-NEXT:     (call $get-i32)
+  ;; CHECK-NEXT:     (call $get-i32)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (call $get-i32)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $restructure-br_if-value-redundant-in-block-tail-3 (param $x i32) (result i32)
+    ;; As above, but now the value has effects, so we do not optimize
+    (block $parent (result i32)
+      (call $nothing)
+      (drop
+        (br_if $parent
+          (call $get-i32)
+          (call $get-i32)
+        )
+      )
+      (call $get-i32)
+    )
+  )
+
+  ;; CHECK:      (func $restructure-br_if-value-redundant-in-block-tail-4 (type $2) (result i32)
+  ;; CHECK-NEXT:  (block $outer (result i32)
+  ;; CHECK-NEXT:   (block $inner (result i32)
+  ;; CHECK-NEXT:    (call $nothing)
+  ;; CHECK-NEXT:    (drop
+  ;; CHECK-NEXT:     (br_if $outer
+  ;; CHECK-NEXT:      (i32.const 1)
+  ;; CHECK-NEXT:      (call $get-i32)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:    (i32.const 1)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $restructure-br_if-value-redundant-in-block-tail-4 (result i32)
+    ;; As above, but the br_if targets another block, so we do not optimize.
+    (block $outer (result i32)
+      (block $inner (result i32)
+        (call $nothing)
+        (drop
+          (br_if $outer
+            (i32.const 1)
+            (call $get-i32)
+          )
+        )
+        (i32.const 1)
+      )
+    )
+  )
+
+  ;; CHECK:      (func $restructure-br_if-value-redundant-in-block-tail-5 (type $2) (result i32)
+  ;; CHECK-NEXT:  (block $parent (result i32)
+  ;; CHECK-NEXT:   (call $nothing)
+  ;; CHECK-NEXT:   (drop
+  ;; CHECK-NEXT:    (br $parent
+  ;; CHECK-NEXT:     (i32.const 1)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (i32.const 1)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $restructure-br_if-value-redundant-in-block-tail-5 (result i32)
+    ;; As above, but the br lacks a condition. We do not bother to optimize
+    ;; the dead code after it, but also should not error here.
+    (block $parent (result i32)
+      (call $nothing)
+      (drop
+        (br $parent
+          (i32.const 1)
+        )
+      )
```

Comment:
```suggestion
      (br $parent
        (i32.const 1)
      )
```
`br` does not need to be dropped: while `br_if` returns a a value in wasm, `br` does not (it is unreachable).